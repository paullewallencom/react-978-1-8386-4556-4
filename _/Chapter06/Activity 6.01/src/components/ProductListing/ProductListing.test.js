import React from "react";
import { render, cleanup } from "@testing-library/react";
import { ProductListing } from "./index";

afterEach(cleanup);

const products = [
  {
    id: 1,
    name: "Expresso",
    price: 5.5,
    summary:
      "Galão, grinder mocha, filter plunger pot siphon mug variety est caramelization. Galão cortado medium latte organic aroma dripper lungo extraction crema. Bar variety spoon a fair trade filter iced. To go mocha wings irish kopi-luwak acerbic doppio kopi-luwak viennese saucer aroma arabica.",
    tags: ["black"]
  },
  {
    id: 2,
    name: "Latte",
    price: 4.5,
    summary:
      "Robusta id fair trade, ristretto froth sugar siphon cream. Rich cinnamon aged espresso carajillo skinny acerbic iced. Froth, blue mountain eu mug, coffee carajillo flavour cappuccino and cortado mocha. Est, blue mountain froth roast as trifecta cappuccino.",
    tags: ["milk"]
  }
];

it("renders listing of products", () => {
  const { asFragment, container } = render(
    <ProductListing products={products} selectedTag={null} />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders filtered number of results", () => {
  const { container } = render(
    <ProductListing products={products} selectedTag="milk" />
  );
  expect(container.querySelectorAll("div.product").length).toBe(1);
});
