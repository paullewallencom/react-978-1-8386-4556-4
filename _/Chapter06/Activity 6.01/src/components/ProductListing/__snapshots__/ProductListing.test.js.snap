// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders listing of products 1`] = `
<DocumentFragment>
  <div>
    <div
      class="product"
    >
      <div
        class="product_header"
      >
        <h2>
          Expresso
        </h2>
        <div>
          <button>
            +
          </button>
        </div>
      </div>
      <strong>
        $5.5
      </strong>
      <div />
    </div>
    <div
      class="product"
    >
      <div
        class="product_header"
      >
        <h2>
          Latte
        </h2>
        <div>
          <button>
            +
          </button>
        </div>
      </div>
      <strong>
        $4.5
      </strong>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
